optuna/integration/_lightgbm_tuner/optimize.py

```diff
@@ -370,6 +370,8 @@ def __init__(
         verbosity: Optional[int] = None,
         show_progress_bar: bool = True,
         model_dir: Optional[str] = None,
+        *,
+        optuna_seed: Optional[int] = None,
```

Comment:
This is just an idea and I want to hear other dev members' opinions, but it might be better to add an option that creates a sampler instance, such as `create_sampler: Optional[Callable[[], optuna.samplers.BaseSampler]]`, instead of `optuna_seed`.
(I think that it's more flexible but it could be too complex than necessary)

Reply:
@sile Thank you for your comment. I agree with your point and your proposal will add more flexibility and a bit complexity. I think at the most flexible case, we could open all IFs to set sampler for each tuner. Currently samplers are hard coded and either grid sampler or TPESampler. I am OK to open all IFs or add flexibility to change only TPESampler part. But if only one IF to set sampler is open, I am wondering if user might assume THE sampler will be used for all tuners..

Reply:
@smly Thank you for your comment. I agree with your concern. Shall I remove the added two tests from this PR? Or it would be appreciated if I could get any other good alternative. Thanks.

Reply:
@tetsuoh0103 Thanks for your response. I understand your concern and now it feels that specifying only the seed is better than allowing an arbitrary sampler.